test/Benchmarks/Metrics/ExemplarBenchmarks.cs

```diff
@@ -33,22 +33,20 @@ public class ExemplarBenchmarks
     private static readonly ThreadLocal<Random> ThreadLocalRandom = new(() => new Random());
     private readonly string[] dimensionValues = new string[] { "DimVal1", "DimVal2", "DimVal3", "DimVal4", "DimVal5", "DimVal6", "DimVal7", "DimVal8", "DimVal9", "DimVal10" };
     private Histogram<long> histogramWithoutTagReduction;
-
     private Histogram<long> histogramWithTagReduction;
-
     private MeterProvider meterProvider;
     private Meter meter;
 
     [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1602:Enumeration items should be documented", Justification = "Test only.")]
-    public enum ExemplarFilterToUse
+    public enum ExemplarConfigurationType
```

Comment:
can you update the benchmark numbers as well? This was added to highlight the relevance of filtering and how perf gets affected when more measurements are considered (not store) for exemplars?

Reply:
I updated the numbers on here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5404/commits/1c75977c82743837ac83b54ca1650b730af89a92

But I will say, I think having the numbers in the source files is bad practice. The numbers are really only useful when run on the same machine and run against other numbers from the same machine run under similar circumstances. To properly benchmark, you need to run numbers on some branch, and then switch to a branch with code changes and re-run them again. Having the numbers in the source file could mislead people into thinking they have made things better or worse unless they correctly re-run the base first 🤔

Reply:
These numbers are useful for tracking memory allocation. It's easy to check if your code adds or reduces the memory allocation by comparing with the results in the source file.

> Having the numbers in the source file could mislead people into thinking they have made things better or worse unless they correctly re-run the base first

We also copy the information about the hardware config when running the benchmarks so if someone is ignoring that, then maybe we could just add a comment saying these numbers are related to that particular config only.

Reply:
Probably not the way to do it. We don't even turn on memory allocation tracking by default you have to add "-m" 🤣 A _solution_ would be more like we mark the things which we expect to be allocation-free and verify that in CI so we aren't reliant on a) people running them and b) people running them correctly understanding the nuances.

Reply:
CI based solution would be much better than what we have for sure.

But in general, for anyone updating benchmarks in a PR, the reviewers check (and should check) if the memory allocation column is added or not. They also check if the PR author ran the benchmarks for both main and the feature branch.

Reply:
we can still keep them with a note above. 
Its very useful to look at it and see how much overhead each feature is adding on.